File: algd2/BinaryTree/src/ch/fhnw/christian/glatthard/algd2/BinaryTree.java
package ch.fhnw.christian.glatthard.algd2;

public class BinaryTree {
	public enum Order {
		PreOrder, InOrder, PostOrder
	};



	private BinaryTreeNode m_root; // Wurzel des Baumes

	// Einfügen eines neuen Schlüssels
	public void insert(Comparable key) {
		if (m_root == null)
			m_root = new BinaryTreeNode(key);
		else
			m_root.insert(key);
	}

	// Baum darstellen
	public void show(Order order) {
		if (m_root == null)
			System.out.println("tree is empty");
		else
			switch (order) {
			case PreOrder:
				System.out.println("pre-order");
				m_root.show(order, 0);
				break;
			case InOrder:
				System.out.println("in-order");
				m_root.show(order, 0);
				break;
			case PostOrder:
				System.out.println("post-order");
				m_root.show(order, 0);
				break;
			default:
				assert false : "wrong traverse order";
			}
	}

}

File: prcpp/BinaryTree/BinaryTree/BinaryTree.cpp
#include "BinaryTree.h"



int main(){
	BinaryTreeNode b = BinaryTreeNode(12);
	std::cout << b.get_value() << std::endl;

	BinaryTreeNode *pb = new BinaryTreeNode(13);
	std::cout << pb->get_value() << std::endl;

	BinaryTree t = BinaryTree(50);
	t.insert(30);
	t.insert(20);
	t.insert(40);
	t.insert(80);
	t.insert(70);
	t.insert(90);
	t.insert(60);
	t.insert(100);
	t.insert(85);
	t.insert(101);
	t.insert(102);
	t.insert(103);
	t.insert(105);

	std::cout << "DEBUG" << std::endl;
	t.remove(30);
	t.remove(20);
	
	
	

	return 0;
	
}